effekt/shared/src/main/scala/effekt/generator/llvm/Transformer.scala

```diff
@@ -110,7 +110,7 @@ object Transformer {
 
       case machine.Switch(value, clauses, default) =>
         emit(Comment(s"switch ${value.name}, ${clauses.length} clauses"))
-        val freeInClauses = clauses.flatMap(freeVariables).toSet ++ default.map(freeVariables).getOrElse(Set.empty)
+        val freeInClauses = clauses.flatMap(freeVariables).toSet ++ default.map(freeVariables).getOrElse(Set.empty) -- Set(value)
```

Comment:
the `-- Set(value)` is semantically wrong here, no?
In the repro (that I just committed... again), there's an `else` clause like `case _ => value` where `value` is also the scrutinee. But the whole problem of this PR was that this pattern seemed to leak. 🔍

Reply:
Yes, I've realized this too. It fixed `tree_explore` though so I wanted to try it on CI.